integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/cache/CarbonDropCacheCommand.scala

```diff
@@ -47,57 +48,35 @@ case class CarbonDropCacheCommand(tableIdentifier: TableIdentifier, internalCall
   def clearCache(carbonTable: CarbonTable, sparkSession: SparkSession): Unit = {
     LOGGER.info("Drop cache request received for table " + carbonTable.getTableName)
 
-    val dropCacheEvent = DropCacheEvent(
-      carbonTable,
-      sparkSession,
-      internalCall
-    )
+    val dropCacheEvent = DropCacheEvent(carbonTable, sparkSession, internalCall)
     val operationContext = new OperationContext
     OperationListenerBus.getInstance.fireEvent(dropCacheEvent, operationContext)
 
     val cache = CacheProvider.getInstance().getCarbonCache
     if (cache != null) {
-      val tablePath = carbonTable.getTablePath + CarbonCommonConstants.FILE_SEPARATOR
-
-      // Dictionary IDs
-      val dictIds = carbonTable.getAllDimensions.asScala.filter(_.isGlobalDictionaryEncoding)
-        .map(_.getColumnId).toArray
+      // Index files
+      val allIndexFiles = CacheUtil.getAllIndexFiles(carbonTable)
+
+      // Dictionary entries
+      val dictIDs: List[String] = carbonTable.getAllDimensions.asScala
+        .filter(_.isGlobalDictionaryEncoding)
+        .map(_.getColumnId).toList
```

Comment:
instead of filter and map use collect like 

`.collect {
        case a if a.isGlobalDictionaryEncoding => 
          a.getColumnId
      }`

Reply:
Done.